Trait select

bmaitner opened the pull request:
@richardjtelford @audhalbritter  Per reviewer comments on the manuscript, I've renamed "trait_impute" to "trait_select" to clarify what these functions are doing (selecting the best available data).  Throughout the code and documentation every instance of impute/imputation/imputed has been replaced with the equivalent select/selection/selected.  I've run the usual checks and tests and I think everyone is working as it should be.

Does this seem ok?

richardjtelford reviewed (changes requested):
Please don't use right assign ->
Run styler on the code to fix spacing etc

richardjtelford reviewed (commented):
I'm not fully convinced by "selected". For tidyverse users, select() is a column operation. It sound to me like it selects traits rather than collating the best values. I'm open to changing from  impute, but I'm not keen on select.

bmaitner commented:
@richardjtelford Good point regarding "select". Do you have a better idea? trait_choose? trait_assign? trait_allocate? I'm also happy with pretty much anything, just need to move away from "impute" so we can say we addressed the reviewer comment.

audhalbritter commented:
> About the trait_impute function, to me it seems that the reviewer did not understand the concept of the trait impute, since he suggests to not include it in the manuscript. So, therefore I think it is rather a matter of explaining that this is not a classical data imputation, but more making use of the hierarchical design. I think that would answer the reviewer comment as well.
Unless we find a great name that we all agree on.

bmaitner commented:
@audhalbritter I see your point, but I think that similar misconceptions might be present in users as well.  The more I think about it, I think "trait_assign" might be relatively clear since the function is basically assigning each species-by-plot combination one or more traits.

richardjtelford commented:
I still like impute. From the font of all knowledge (wikipedia)

"In statistics, imputation is the process of replacing missing data with substituted values."

Which seem to reflect what we are doing. 

I also thought of 
 
 - trait_complete()
 - trait_fill()

Which sort of describe what we are doing without much baggage.

The GitHub actions need updating - I'll fix that

bmaitner commented:
I'd be happy with either Trait_fill or Trait_complete, since, as you say, they both get away from the baggage of  the word "impute".  I think I have a slight preference for trait_fill, but I'd be happy with either if you and/or @audhalbritter have a preference.
